src/gtk/caret.cpp

```diff
@@ -0,0 +1,241 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/caret.cpp
+// Purpose:     GTK3 implementation of wxCaret
+// Author:      Kettab Ali
+// Created:     2022-01-29
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if wxUSE_CARET && defined(__WXGTK3__)
+
+#ifndef WX_PRECOMP
+    #include "wx/window.h"
+#endif // WX_PRECOMP
+
+#include "wx/caret.h"
+#include "wx/gtk/private/wrapgtk.h"
+
+// ===========================================================================
+// implementation
+// ===========================================================================
+
+// ---------------------------------------------------------------------------
+// blink time
+// ---------------------------------------------------------------------------
+
+//static
+int wxCaretBase::GetBlinkTime()
+{
+    GtkSettings* settings = gtk_settings_get_default();
+
+    GValue blink = G_VALUE_INIT;
+    g_value_init(&blink, G_TYPE_BOOLEAN);
+    g_object_get_property(G_OBJECT(settings), "gtk-cursor-blink", &blink);
+
+    if ( !g_value_get_boolean(&blink) )
+        return 0;
+
+    GValue blinkTime = G_VALUE_INIT;
+    g_value_init(&blinkTime, G_TYPE_LONG);
+    g_object_get_property(G_OBJECT(settings), "gtk-cursor-blink-time", &blinkTime);
+
+    // Quoting gtk-cursor-blink-time documentation:
+    //
+    //   "Length of the cursor blink cycle, in milliseconds.
+    //    Default value: 1200"
+    //
+    // blinkTime (blink cycle) = time interval between 2 consecutive blinks.
+    //
+    return g_value_get_long(&blinkTime) / 2;
+}
+
+//static
+void wxCaretBase::SetBlinkTime(int milliseconds)
+{
+    GtkSettings* settings = gtk_settings_get_default();
+
+    GValue blink = G_VALUE_INIT;
+    g_value_init(&blink, G_TYPE_BOOLEAN);
+
+    if ( milliseconds == 0 )
+    {
+        g_value_set_boolean(&blink, FALSE);
+    }
+    else
+    {
+        g_value_set_boolean(&blink, TRUE);
+
+        GValue blinkTime = G_VALUE_INIT;
+        g_value_init(&blinkTime, G_TYPE_LONG);
+        g_value_set_long(&blinkTime, 2 * milliseconds);
+        g_object_set_property(G_OBJECT(settings), "gtk-cursor-blink-time", &blinkTime);
+    }
+
+    g_object_set_property(G_OBJECT(settings), "gtk-cursor-blink", &blink);
+}
+
+// ---------------------------------------------------------------------------
+// wxCaret implementation
+// ---------------------------------------------------------------------------
+
+void wxCaret::Init()
+{
+    m_hasFocus = true;
+    m_blinkedOut = true;
+
+    m_blinkTime = GetBlinkTime();
+}
+
+void wxCaret::SetupTimer()
+{
+    wxWindow* const win = GetWindow();
+    m_timer.SetOwner(win);
+
+    win->Bind(wxEVT_TIMER, &wxCaret::OnTimer, this);
```

Comment:
Is it possible to destroy/remove a `wxCaret`? Based on `wxWindowBase::SetCaret` I would think so.
Then you should unbind the event from the window in the caret's destructor. Otherwise the window will not be aware that the caret has been destroyed and call `OnTimer` on an invalid object.